File: atom-backend/atom-service/src/main/java/cn/iinti/atom/service/base/metric/mql/func/FuncGetVar.java
package cn.iinti.atom.service.base.metric.mql.func;

import cn.iinti.atom.service.base.metric.mql.Context;

import java.util.List;

@MQLFunction.MQL_FUNC("getVar")
public class FuncGetVar extends MQLFunction {
    private final String varName;

    public FuncGetVar(List<String> params) {
        super(params);
        varName = params.get(0);
    }

    @Override
    public Context.MQLVar call(Context context) {
        Context.MQLVar mqlVar = context.getVariables().get(varName);
        if (mqlVar == null) {
            return null;
        }
        return mqlVar.copy();
    }
}

File: server/src/main/kotlin/cn/iinti/katom/base/metric/mql/func/FuncGetVar.kt
package cn.iinti.katom.base.metric.mql.func

import cn.iinti.katom.base.metric.mql.Context
import cn.iinti.katom.base.metric.mql.Context.MQLVar
import cn.iinti.katom.base.metric.mql.func.MQLFunction.MQL_FUNC


@MQL_FUNC("getVar")
class FuncGetVar(params: List<String>) : MQLFunction(params) {
    private val varName = params[0]

    override fun call(context: Context): MQLVar? {
        val mqlVar = context.variables[varName] ?: return null
        return mqlVar.copy()
    }
}
